dnf-behave-tests/features/builddep.feature

```diff
@@ -80,3 +80,62 @@ Scenario: Builddep on SPEC with non-available Source0
 
    Error: Some packages could not be found.
    """
+
+@bz1758459
+Scenario: I can exclude package version and install the best candidate
+  Given I use repository "dnf-ci-fedora"
+    And I enable plugin "builddep"
+  Given I create file "dummy-pkg.spec" with
+   """
+   Name: dummy-pkg
+   Summary: dummy-pkg summary
+   Version: 1.0
+   Release: 1
+   License: GPL
+   BuildRequires: nodejs
+   %description
+   This is a dummy-pkg description
+   %build
+   %files
+   %changelog
+   """
+   When I execute dnf with args "builddep {context.dnf.installroot}/dummy-pkg.spec --exclude nodejs-1:5.12.1-1.fc29 --best"
+   Then the exit code is 0
+    And Transaction contains
+        | Action                | Package                       |
+        | install               | nodejs-1:5.11.1-1.fc29.x86_64 |
```

Comment:
Please use the exact match version of this step (instead of `Transaction contains`):
```gherkin
   And Transaction is following
   ...
```

(You'll need to list all the packages in the transaction)

We're planning on deprecating the partial transaction check step, as there is no reason for it (except for being a bit less verbose), checking the exact output is always better in tests.